Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTTStub.cc

```diff
@@ -81,48 +128,279 @@ Phase2OTValidateTTStub::~Phase2OTValidateTTStub() {
+      } else {
+        edm::LogVerbatim("Tracklet") << "WARNING -- neither TOB nor TID stub, shouldn't happen...";
+        layer = -1;
+      }
+
+      int isPSmodule = (topo.nrows() == 960) ? 1 : 0;
+
+      // Calculate local coordinates of clusters
+      MeasurementPoint innerClusterCoords = tempStubPtr->clusterRef(0)->findAverageLocalCoordinatesCentered();
+      MeasurementPoint outerClusterCoords = tempStubPtr->clusterRef(1)->findAverageLocalCoordinatesCentered();
+
+      // Convert local coordinates to global positions
+      Global3DPoint innerClusterGlobalPos = theGeomDet->surface().toGlobal(theGeomDet->topology().localPosition(innerClusterCoords));
+      Global3DPoint outerClusterGlobalPos = theGeomDet->surface().toGlobal(theGeomDet->topology().localPosition(outerClusterCoords));
+
+      // Determine maximum Z positions of the stubs
+      float stub_maxZ = std::max(innerClusterGlobalPos.z(), outerClusterGlobalPos.z());
+
+      // Stub parameters
+      float stub_phi = innerClusterGlobalPos.phi();
+      float stub_r = innerClusterGlobalPos.perp();
+      float stub_z = innerClusterGlobalPos.z();
+
+      // Tracking particle parameters
+      int myTP_charge = my_tp->charge();
+      float myTP_pt = my_tp->p4().pt();
+      float myTP_eta = my_tp->p4().eta();
+
+      if (myTP_charge == 0) continue;
+      if (myTP_pt < TP_minPt) continue;
+      if (std::abs(myTP_eta) > TP_maxEta) continue;
+
+      // Derived coordinates
+      std::vector<double> tpDerivedCoords = getTPDerivedCoords(my_tp, isBarrel, modMaxZ, modMinZ, stub_r);
+      float tp_z = tpDerivedCoords[0];
+      float tp_phi = tpDerivedCoords[1];
+      
+      // Trigger information
+      float trigBend = tempStubPtr->bendFE();
+      if (!isBarrel && stub_maxZ < 0.0) {
+        trigBend = -trigBend;
+      }
+
+      float correctionValue = phiOverBendCorrection(isBarrel, stub_z, stub_r, tTopo_, detid, det0, det1);
+      float trackBend = -(sensorSpacing * stub_r * bfield_ * c_ * myTP_charge) /
+                        (stripPitch * 2.0E2 * myTP_pt * correctionValue);
+
+      float bendRes = trackBend - trigBend;
+      float zRes = tp_z - stub_z;
+      float phiRes = tp_phi - stub_phi;
+      
+      // Fill histograms
+      if (Stub_RZ) {
+        Stub_RZ->Fill(stub_z, stub_r);
+      } else {
+        edm::LogError("Phase2OTValidateTTStub") << "Error: Stub_RZ histogram is null";
+      }
+      // Histograms for z_res and r_res based on module type and location
+      if (isBarrel == 1) {
+          if (isPSmodule) {
+              z_res_isPS_barrel->Fill(zRes);
+              phi_res_isPS_barrel->Fill(phiRes);
+          } else {
+              z_res_is2S_barrel->Fill(zRes);
+              phi_res_is2S_barrel->Fill(phiRes);
+          }
+      }
+
+      // Ensure that the vectors are correctly assigned before use
+      if (isBarrel == 1) {
+          phi_res_discs = &phi_res_barrel_layers; // Pointing to barrel layers vector
```

Comment:
What is going on here? If we're in the barrel, why is variable phi_res_discs being set?